src/core/transport/secure_endpoint_transport.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2025 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_SRC_CORE_TRANSPORT_SECURE_ENDPOINT_TRANSPORT_H
+#define GRPC_SRC_CORE_TRANSPORT_SECURE_ENDPOINT_TRANSPORT_H
+
+#include "src/core/client_channel/client_channel_factory.h"
+#include "src/core/client_channel/subchannel.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/util/no_destruct.h"
+
+namespace grpc_core {
+
+namespace secure_endpoint_transport_detail {
+
+class GenericClientChannelFactory : public ClientChannelFactory {
+ public:
+  RefCountedPtr<Subchannel> CreateSubchannel(
+      const grpc_resolved_address& address, const ChannelArgs& args) override {
+    absl::StatusOr<ChannelArgs> new_args = GetSecureNamingChannelArgs(args);
+    if (!new_args.ok()) {
+      LOG(ERROR) << "Failed to create channel args during subchannel creation: "
+                 << new_args.status() << "; Got args: " << args.ToString();
+      return nullptr;
+    }
+    RefCountedPtr<Subchannel> s =
```

Comment:
Nit: Doesn't seem like we need this temp variable.

Reply:
fixed; also moved this method into the cc